tests/Games/NexusMods.Games.FOMOD.Tests/FomodXmlInstallerTests.cs

```diff
@@ -137,6 +137,44 @@ public async Task InstallsFilesSimple_Using7z()
                 x => x.To.FileName == "g2p1f1.out.esp"
             );
     }
+    
+    [Fact]
+    public async Task InstallFilesNestedWithImages()
+    {
+        using var testData = await SetupTestFromDirectoryAsync("NestedWithImages.zip");
+        var installedFiles = (await testData.GetFilesToExtractAsync()).ToArray();
+
+        installedFiles
+            .Cast<IToFile>()
```

Comment:
You should verify that the items are of type `IToFile`:

```csharp
installedFiles
  .Should()
  .AllBeOfType<IToFile>()
  .Which
  .Should()
  //... continue with have count
```

Reply:
`AllBeOfType<IToFile>()` fails the assertion saying that the found type was `ModFiles.FromArchive`, which in turn implements the `IToFile` interface. It doesn't make much sense

Reply:
Right, looking at the docs, `AllBeOfType` checks for the **exact** type, which isn't very helpful in this scenario.

Reply:
`AllBeAssignableTo<>()` works